webapp/src/components/expanded_view/component.tsx

```diff
@@ -632,6 +675,36 @@ export default class ExpandedView extends React.PureComponent<Props, State> {
         });
     }
 
+    renderRecordingBadge = () => {
+        // This should not render if:
+        // - The recording has not been initialized yet OR if it has ended.
+        if (!this.props.callRecording?.init_at || this.props.callRecording?.end_at) {
+            return null;
+        }
+
+        const isHost = this.props.callHostID === this.props.currentUserID;
+        const hasRecStarted = this.props.callRecording?.start_at;
+
+        // If the recording has not started yet then we only render if the user
+        // is the host, in which case we'll show the loading spinner.
+        if (!isHost && !hasRecStarted) {
+            return null;
+        }
+
+        return (
+            <Badge
+                text={'REC'}
+                textSize={12}
+                gap={6}
+                margin={'0 12px 0 0'}
+                padding={'8px 6px'}
+                icon={(<RecordCircleOutlineIcon size={12}/>)}
+                bgColor={hasRecStarted ? '#D24B4E' : 'rgba(221, 223, 228, 0.04)'}
+                loading={!hasRecStarted}
```

Comment:
ah, I see -- we want to send out the call recording ws event, and then send out the `start_at` after, so we have UX feedback while it's getting set up. Gotcha.

Reply:
Yeah, since it's clearly an async operation I wanted to get that kind of feedback and potentially error out as well. But I'll review all this once I have the service logic plugged in.